src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/AwsSdk/SqsHelper.cs

```diff
@@ -61,17 +62,19 @@ public static ISegment GenerateSegment(ITransaction transaction, MethodCall meth
             return segment;
         }
 
-        public static void InsertDistributedTraceHeaders(ITransaction transaction, object sendMessageRequest)
+        public static void InsertDistributedTraceHeaders(ITransaction transaction, object sendMessageRequest, IEnumerable<string> dtHeaders)
         {
             var headersInserted = 0;
 
+            var requiredHeadroom = dtHeaders.Count();
+
             var setHeaders = new Action<object, string, string>((smr, key, value) =>
             {
                 var getMessageAttributes = _getMessageAttributes.GetOrAdd(smr.GetType(), t   => VisibilityBypasser.Instance.GeneratePropertyAccessor<IDictionary>(t, "MessageAttributes"));
                 var messageAttributes = getMessageAttributes(smr);
 
                 // SQS is limited to no more than 10 attributes; if we can't add up to 3 attributes, don't add any
```

Comment:
Maybe we should update the comment since it isn't necessarily `3`? And maybe make `10` a `const` so we remember not to mess with it?